storage/rocksdb/ha_rocksdb.cc

```diff
@@ -10414,15 +10467,33 @@ int ha_rocksdb::inplace_populate_sk(
       }
 
       /* Create new secondary index entry */
+
+      String value_buffer;
```

Comment:
`pack_record` will sometimes clear the passed `Rdb_string_writer` if `unpack_info` only contains the header. This makes things tricky since `Rdb_string_writer` doesn't provide any way to clear only a part of the data. One way to avoid using a buffer is to write the index flags, call `pack_record`, and write the index flags again if the writer was cleared by `pack_record`. Thoughts?